assets/js/blocks/stock-filter/block.tsx

```diff
@@ -385,18 +418,104 @@ const StockStatusFilterBlock = ( {
 		<>
 			{ ! isEditor && blockAttributes.heading && filterHeading }
 			<div
-				className={ classnames( 'wc-block-stock-filter', {
-					'is-loading': isLoading,
-				} ) }
+				className={ classnames(
+					'wc-block-stock-filter',
+					`style-${ blockAttributes.displayStyle }`,
+					{
+						'is-loading': isLoading,
+					}
+				) }
 			>
-				<CheckboxList
-					className={ 'wc-block-stock-filter-list' }
-					options={ displayedOptions }
-					checked={ checked }
-					onChange={ onChange }
-					isLoading={ isLoading }
-					isDisabled={ isDisabled }
-				/>
+				{ blockAttributes.displayStyle === 'dropdown' ? (
+					<>
+						<FormTokenField
+							key={ remountKey }
+							className={ classnames( borderProps.className, {
+								'single-selection': ! multiple,
+								'is-loading': isLoading,
+							} ) }
+							style={ {
+								...borderProps.style,
+								borderStyle: 'none',
+							} }
+							suggestions={ displayedOptions
+								.filter(
+									( option ) =>
+										! checked.includes( option.value )
+								)
+								.map( ( option ) => option.value ) }
+							disabled={ isLoading }
+							placeholder={ __(
+								'Select stock status',
+								'woo-gutenberg-products-block'
+							) }
+							onChange={ ( tokens: string[] ) => {
+								if ( ! multiple && tokens.length > 1 ) {
+									tokens = [ tokens[ tokens.length - 1 ] ];
+								}
+
+								tokens = tokens.map( ( token ) => {
+									const displayOption = displayedOptions.find(
+										( option ) => option.value === token
+									);
+
+									return displayOption
+										? displayOption.value
+										: token;
+								} );
+
+								const added = difference( tokens, checked );
+
+								if ( added.length === 1 ) {
```

Comment:
> What happens if .length is > 1?

To be honest, I was trying to find a case when the `.length` is > 1 and couldn't find it. But that check (`if ( added.length === 1 )` ) is there in Filter by Attribute and I assumed it's there for a reason and there may be a case I'm not aware of.

> Or that can never happen because it is assumed that onChange is called at every selection?

I think that's actually the point. The only thing I could think of was that the rest of the logic is not prepared for more than one token difference at once (e.g. screen reader communicates a single token was added)

> What happens if someone copy+pastes a string with multiple valid tokens into the input?

User has to explicitly confirm the selection of each token, so they can only copy-paste the search string, not the token itself. Hence, I think it's impossible to add more than single token at once. Unless you know such case? I'm open for suggestions here.